lib/utils/client_factory.js

```diff
@@ -0,0 +1,20 @@
+var pooling = require('./polling_client');
+var socketio = require('./socket_client');
+
+module.exports = function factory(options) {
```

Comment:
export an object with a function `create` or something like that instead of the function directly. otherwise client_factory is not a good name